src/wallet/wallet.cpp

```diff
@@ -4182,8 +4182,8 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(std::shared_ptr<CWallet>
 
         // Make list of wallets to cleanup
         std::vector<std::shared_ptr<CWallet>> created_wallets;
-        created_wallets.push_back(std::move(res.watchonly_wallet));
-        created_wallets.push_back(std::move(res.solvables_wallet));
+        if (res.watchonly_wallet) created_wallets.push_back(std::move(res.watchonly_wallet));
+        if (res.solvables_wallet) created_wallets.push_back(std::move(res.solvables_wallet));
```

Comment:
`res` is returned by the function, couldn't a move operation introduce UB here? 
```suggestion
        if (res.watchonly_wallet) created_wallets.push_back(res.watchonly_wallet);
        if (res.solvables_wallet) created_wallets.push_back(res.solvables_wallet);
```

Reply:
`res` isn't returned by the failure path. These also need to be moved in order to keep the refcount at 1 so that these wallets are properly removed and cleaned up in this failure path.

Reply:
Good point, thanks. Resolved.